include/openthread/ip6.h

```diff
@@ -223,6 +223,7 @@ typedef struct otMessageInfo
                                  ///< Otherwise, specifies the IPv6 Hop Limit.
     bool mIsHostInterface : 1;   ///< TRUE if packets sent/received via host interface, FALSE otherwise.
     bool mAllowZeroHopLimit : 1; ///< TRUE to allow IPv6 Hop Limit 0 in `mHopLimit`, FALSE otherwise.
+    bool mMulticastLoop : 1;     ///< TRUE to allow looping back mutlicast, FALSE otherwise.
```

Comment:
With this we allow users to select which msgs would allow loop back and which won't.
Can we consider allowing it for all msgs ? Any side-effect/reason not to? Thanks.
Or can we somehow deduce this in another way? Thanks.

Reply:
If we allow all, this would increase overhead to all nodes sending multicast messages, including Thread protocol messages (MLE advertisements, link requests, address queries). POSIX also provides a flag controlling this behavior but per socket instead of per message.

Reply:
Allowing all would mean a significant change to the current OpenThread behavior. Might have some unexpected issues.